mozilla/blurts-server
# MNTOR-2224 - allow updating FxA profile info from front-end without sign-out

# References: 
Jira: MNTOR-2224



# Description

This causes `update()` from `useSession` to trigger a profile refresh, to update the list of subscriptions (and any other info that has changed). It should not require the user to re-authenticate, as it uses the stored access token.

As noted in a comment in the code, this makes use of a client-side function as there doesn't seem to be any supported way to do this server-side in Next-Auth. I experimented a bit with this and I think we could do it with some effort, but this i a far smaller change (at the cost of an extra redirect to the user).

# How to test

Positive test:
1. create a new account
2. go through subscription flow
3. observe redirect from welcome-to-premium -> subscription -> back

Negative test:
1. try to go directly to welcome-to-premium page without having subscribed
2. observe redirect to / (landing page)

# Checklist (Definition of Done)
- [x] Remove demo button
